packages/default-reporter/src/reporterForClient/reportProgress.ts

```diff
@@ -12,28 +14,32 @@ type ProgressStats = {
 
 type ModulesInstallProgress = {
   importingDone$: most.Stream<boolean>,
+  packageImportMethod$: most.Stream<string>,
   progress$: most.Stream<ProgressStats>,
   requirer: string,
 }
 
 export default (
   log$: {
     progress: most.Stream<ProgressLog>,
+    packageImportMethod: most.Stream<PackageImportMethodLog>,
     stage: most.Stream<StageLog>,
   },
   opts: {
     cwd: string,
     throttleProgress?: number,
+    modulesDir?: string,
   }
 ) => {
   const progressOutput = typeof opts.throttleProgress === 'number' && opts.throttleProgress > 0
     ? throttledProgressOutput.bind(null, opts.throttleProgress)
     : nonThrottledProgressOutput
 
-  return getModulesInstallProgress$(log$.stage, log$.progress)
-    .map(({ importingDone$, progress$, requirer }) => {
-      const output$ = progressOutput(importingDone$, progress$)
-
+  return getModulesInstallProgress$(log$.stage, log$.progress, log$.packageImportMethod)
+    .map(({ importingDone$, progress$, packageImportMethod$, requirer }) => {
+      const nodeModulesPath = path.join(opts.modulesDir ?? requirer, 'node_modules')
+      const showImportingMethod = !fs.existsSync(nodeModulesPath)
```

Comment:
it would be nice to avoid this fs operation altogether. We could maybe add some log for currentLockfile exists.

That is info is available here: https://github.com/pnpm/pnpm/blob/98ed2544d28582fae83f983eef796365dddc6a01/packages/get-context/src/readLockfiles.ts#L81

if the current lockfile does not exist, it means  there is no node_modules.

Reply:
the name is bit confusing to me: does it check existing lockfile? it's possible that lockfile exists but node_modules don't